### DISCUSSION TITLE
playlist tabs vs. window size on macos sonoma

### DISCUSSION THREAD

**Discussion by wrcarithers** [audacious-media-player/audacious#1455] [Q&A] [answered]
I'm having an interesting problem with a fresh install of Audacious 4.4 on a new MacBook Air running Sonoma (14.6.1). The issue is that the width of the Audacious player window is affected by the number of playlists I create.  I'm using the Qt interface.

I first noticed this when I had created 48 playlists (per-CD, from my collection) and the player window grew to roughly 2.5 times the width of my laptop screen. Through experimentation, I discovered that this seems to be due to Audacious wanting to create a playlist tab for each playlist, and forcing the window to be wide enough to display all the tabs at once, even though I have the "playlist tab" display setting at "Never". Although I can make the window wider, I cannot shrink it - the minimum width is based on the number of playlists even though I don't show the tabs.

I tried removing all playlists and deleting all the config files, then running Audacious. I set the tab display setting to "Never", and began adding playlists one at a time. Sure enough, once I had added enough playlists, adding one more caused the window to expand enough to hold a couple of additional tabs.

I went into the config files and tried manually setting the window width to a smaller value (e.g., it was 2653, and I dropped it to 653). Starting up Audacious expanded the window to the full width, and exiting caused it to rewrite the config file with the larger width value.

I also experimented with the "show entry counts" and "show close buttons" settings, but even with both of those unchecked and not showing tabs, the minimum width is determined by the number of playlists.

Is this the expected behavior, or is there some sort of glitch in my installation? I can't find any setting that seems to affect this beyond the three I described above; is there one that I'm missing?

**Comment by radioactiveman** [comment]
> Is this the expected behavior, or is there some sort of glitch in my installation?

With the "Show playlist tabs" setting "Never" I would also expect that the window size is not affected by the number of playlists.

I have tried to reproduce this on Linux but here the window size stays the same, even with 100 playlists.
Does macOS show arrow buttons beside the playlist tabs if they don't fit the window size? Of course if "Show playlist tabs" is enabled.

Maybe we need to enable `usesScrollButtons` on our side if it is false on macOS.
https://doc.qt.io/qt-6/qtabbar.html#usesScrollButtons-prop

```diff
diff --git a/src/qtui/playlist_tabs.cc b/src/qtui/playlist_tabs.cc
index d23952108..564f1915e 100644
--- a/src/qtui/playlist_tabs.cc
+++ b/src/qtui/playlist_tabs.cc
@@ -221,6 +221,7 @@ PlaylistTabBar::PlaylistTabBar(QWidget * parent) : QTabBar(parent)
 {
     setMovable(true);
     setDocumentMode(true);
+    setUsesScrollButtons(true);
     updateSettings();
 
     connect(this, &QTabBar::tabMoved, this, &PlaylistTabBar::tabMoved);

```

**Reply by wrcarithers** [reply]
I set "Show playlist tabs" to "Always" and the tabs appeared, but there were no arrows at either end of the tab list. The tabs themselves shrink down to a minimal size that seems to be based on the length of the playlist name, but all of them are displayed, with the "active" playlist's tab having an icon indicating that it's playing or is paused. Definitely looks like it's not scrolling the tabs.

HOWEVER, I tried switching to the "Dark" theme, and was surprised to see arrow buttons appear! I don't use a dark theme normally, so  I hadn't tried that before. With the Dark theme in use, the problems I've had disappear - I can resize the window horizontally to any width I like. It also doesn't shrink the tabs - the full playlist titles appear.

Another difference with the Dark theme is that I can dock any of the additional widgets (lyrics, playlist manager, etc.) at the left or the right side of the playlist widget. I could not do that with the Native theme; I could only dock things above or below the playlist.

So, I guess the question now is "What's different about the Dark theme compared to the Native theme that causes these changes?"

**Comment by radioactiveman** [comment, marked as answer]
The window size issue is fixed now for Audacious 4.4.1 with https://github.com/audacious-media-player/audacious-plugins/commit/db4ae760a096bc72a3be10ea803d3ce733fc78e3. Thanks for reporting this.

**Comment by wrcarithers** [comment]
Thanks for the quick work on fixing the problem!
